### PR TITLE
[python] Return fig, ax

### PR DESCRIPTION
**Pull request by pluegh**
Enables further manipulation of the figure after creating a plot.

### CODE REVIEW

**Comment by mreboud** [comment]
Despite the simple workaround, I think that this PR is still needed, do you agree @dvandyk @pluegh?

**Comment by mreboud** [comment]
@pluegh Didn't you say you wanted to adapt the examples notebooks?

**Comment by dvandyk** [comment]
> Despite the simple workaround, I think that this PR is still needed, do you agree @dvandyk @pluegh?

Yes please!

File: python/eos/plot/plotter.py [hunk 1]
```diff
@@ -1961,7 +1961,9 @@ def plot_contents(self):
 
 
     def plot(self):
-        """Produces the plot"""
+        """Produces the plot
+
+        returns fig, ax from matplotlib.pyplot.subplots()"""
```
Comment (dvandyk) [inline python/eos/plot/plotter.py:1966]: Can we please use the Sphinx syntax for documenting return values?

File: python/eos/plot/plotter.py [hunk 2]
```diff
@@ -1970,6 +1972,8 @@ def plot(self):
         if self.output:
             plt.savefig(self.output, bbox_inches='tight', dpi=300)
 
+        return self.fig, self.ax
```
Comment (dvandyk) [inline python/eos/plot/plotter.py:1975]: OK for now. Will need to revisit if we break API and permit more than one axis per plot.

**Review by dvandyk — CHANGES_REQUESTED** [review]

**Comment by pluegh** [comment]
> Despite the simple workaround, I think that this PR is still needed, do you agree @dvandyk @pluegh?

I don't think it's necessary as it's documented in the user reference and an "advanced" feature that would clutter the notebooks.

File: python/eos/plot/plotter.py (hunk 1 above)
Comment (pluegh) [reply python/eos/plot/plotter.py]: Thanks for the suggestion, I changed it and tested it in sphinx.

File: python/eos/plot/plotter.py (hunk 2 above)
Comment (pluegh) [reply python/eos/plot/plotter.py:1978]: Indeed.

**Review by dvandyk — APPROVED** [review]

**Comment by mreboud** [comment]
I still don't see the modification to the notebooks, so this will 'break' the documentation

**Pull request merged**